src/analysis/bounded_variation.lean

```diff
@@ -537,66 +537,63 @@ end
 lemma comp_le_of_monotone_on (f : α → E) {s : set α} {t : set β} (φ : β → α)
   (hφ : monotone_on φ t) (φst : set.maps_to φ t s) :
   evariation_on (f ∘ φ) t ≤ evariation_on f s :=
-begin
-  apply supr_le _,
-  rintro ⟨n, ⟨u, hu, ut⟩⟩,
-  exact le_supr (λ (p : ℕ × {u : ℕ → α // monotone u ∧ ∀ i, u i ∈ s}),
-    ∑ i in finset.range p.1, edist (f ((p.2 : ℕ → α) (i+1))) (f ((p.2 : ℕ → α) i)))
-    ⟨n, ⟨φ ∘ u, λ x y xy, hφ (ut x) (ut y) (hu xy), λ i, φst (ut i)⟩⟩,
-end
+supr_le $ λ ⟨n, u, hu, ut⟩, le_supr_of_le
+  ⟨n, φ ∘ u, λ x y xy, hφ (ut x) (ut y) (hu xy), λ i, φst (ut i)⟩ le_rfl
 
 lemma comp_le_of_antitone_on (f : α → E) {s : set α} {t : set β} (φ : β → α)
   (hφ : antitone_on φ t) (φst : set.maps_to φ t s) :
   evariation_on (f ∘ φ) t ≤ evariation_on f s :=
-begin
-  apply supr_le _,
-  rintros ⟨n, ⟨u, hu, ut⟩⟩,
-  change ∑ i in finset.range n, edist (f ∘ φ $ u (i+1)) (f ∘ φ $ u i) ≤ evariation_on f s,
+supr_le $ λ ⟨n, u, hu, ut⟩, begin
```

Comment:
It's generally more readable to have a proof which is easier full term mode or full tactic mode. Can you go full tactic mode here, starting with `refine supr_le _`, `rintros ⟨n, u, hu, ut⟩` (untested)?

Reply:
Done! Also golfed `sum_le` using the same `le_supr_of_le` trick.